src/Microsoft.AspNet.Mvc.Core/Filters/DefaultFilterProvider.cs

```diff
@@ -31,6 +34,10 @@ public void OnProvidersExecuting([NotNull] FilterProviderContext context)
                 {
                     ProvideFilter(context, item);
                 }
+                context.Results =
+                    context.Results
+                    .OrderBy(filter => filter.Descriptor, FilterDescriptorCorsComparer.Comparer)
```

Comment:
I understand the scenario of more filter providers, but I really do not see a scenario where someone where smart needs to tweak this part. We discussed having a high priority order but that was shot down as it is easier to change. 
Absolute guarantee can be provided only by a new filter stage and we did not want that.